### PR TITLE
update vw version

### PR DESCRIPTION
**Pull request by lalo**
- brings around 63 more commits from VW
relevant to this change:
- PR #2890 from VW (is_newline change)
- PR #2925 from VW (dealloc example)
- some changes around pointers to unique_ptrs
- brings vw 8.10.1 + some new changes

### CODE REVIEW

**Comment by lalo** [comment]
2890

File: external_parser/example_joiner.cc [hunk 1]
```diff
@@ -466,6 +466,7 @@ int example_joiner::process_joined(v_array<example *> &examples) {
     // add an empty example to signal end-of-multiline
     examples.push_back(&VW::get_unused_example(_vw));
     _vw->example_parser->lbl_parser.default_label(&examples.back()->l);
+    examples.back()->is_newline = true;
```
Comment (lalo) [inline external_parser/example_joiner.cc:469]: is this the right place?

Comment (lalo) [reply external_parser/example_joiner.cc:469]: https://github.com/VowpalWabbit/vowpal_wabbit/pull/2890/files

Comment (olgavrou) [reply external_parser/example_joiner.cc:469]: yup

**Review by olgavrou — APPROVED** [review]
LGTM

Maybe add a small list in the description of the PR with the most important changes in vw that are being pulled in? (e.g. https://github.com/VowpalWabbit/reinforcement_learning/pull/279)

**Review by jackgerrits — APPROVED** [review]

**Pull request merged**